packages/webapp/pages/squads/[handle]/moderate.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { ReactElement, useEffect } from 'react';
+import {
+  ManageSquadPageContainer,
+  SquadSettingsProps,
+} from '@dailydotdev/shared/src/components/squads/utils';
+import {
+  SquadTab,
+  SquadTabs,
+} from '@dailydotdev/shared/src/components/squads/SquadTabs';
+import { SquadModerationList } from '@dailydotdev/shared/src/components/squads/moderation/SquadModerationList';
+import {
+  PageHeader,
+  PageHeaderTitle,
+} from '@dailydotdev/shared/src/components/layout/common';
+import {
+  Button,
+  ButtonVariant,
+} from '@dailydotdev/shared/src/components/buttons/Button';
+import { ArrowIcon } from '@dailydotdev/shared/src/components/icons';
+import {
+  GetStaticPathsResult,
+  GetStaticPropsContext,
+  GetStaticPropsResult,
+} from 'next';
+import { ParsedUrlQuery } from 'querystring';
+import { useSquad } from '@dailydotdev/shared/src/hooks';
+import { useRouter } from 'next/router';
+import { getLayout as getMainLayout } from '../../../components/layouts/MainLayout';
+
+export default function ModerateSquadPage({
+  handle,
+}: SquadSettingsProps): ReactElement {
+  const { squad, isFetched } = useSquad({ handle });
+  const router = useRouter();
+
+  useEffect(() => {
+    if (!isFetched) {
+      return;
+    }
+
+    if (squad?.moderationRequired) {
+      return;
+    }
+
+    router.push(squad ? squad.permalink : '/404');
+  }, [squad, isFetched, router]);
+
+  return (
+    <ManageSquadPageContainer>
+      <PageHeader className="border-b-0">
+        <Button
+          variant={ButtonVariant.Tertiary}
+          icon={<ArrowIcon className="-rotate-90" />}
+        />
+        <PageHeaderTitle className="typo-title3">
+          Squad settings
+        </PageHeaderTitle>
+      </PageHeader>
+      <SquadTabs
+        active={SquadTab.PendingPosts}
+        handle={handle}
+        pendingCount={squad?.moderationPostCount}
+      />
+      <SquadModerationList squad={squad} />
+    </ManageSquadPageContainer>
+  );
+}
+
+export async function getStaticPaths(): Promise<GetStaticPathsResult> {
+  return { paths: [], fallback: true };
+}
+
+interface SquadPageParams extends ParsedUrlQuery {
+  handle: string;
+}
+
+export function getStaticProps({
```

Comment:
We could probably load the squad server side here?

Reply:
Thought of that too intially, but what data will we load from there?

Reply:
You can choose what data you need basically.
Whatever you need to render the page should be there in that case.

Reply:
Yep. Also moved the redirection parts on the server side 👀